pkg/agent/openflow/network_policy.go

```diff
@@ -598,54 +695,161 @@ func (c *client) InstallPolicyRuleFlows(rule *types.PolicyRule) error {
 		if err := c.flowOperations.AddAll(actionFlows); err != nil {
 			return nil
 		}
+		// Add the action flows after the Openflow entries are installed on the OVS bridge successfully.
 		conj.actionFlows = actionFlows
 	}
+	c.conjMatchFlowLock.Lock()
+	defer c.conjMatchFlowLock.Unlock()
+
+	// Calculate the conjMatchFlowContext changes. The changed Openflow entries are included in the conjMatchFlowContext change.
+	ctxChanges := conj.getConjunctiveMatchFlowAddChange(c, rule)
+
+	// Send the changed Openflow entries to the OVS bridge, and then update the conjMatchFlowContext as the expected status.
+	if err := c.installConjunctiveMatchFlows(ctxChanges); err != nil {
+		return err
+	}
+	// Add the policyRuleConjunction into policyCache.
+	c.policyCache.Store(rule.ID, conj)
+	return nil
+}
+
+// installConjunctiveMatchFlows installs Openflow entries on the OVS bridge, and then updates the conjMatchFlowContext.
+// Release the lock in the conjMatchFlowContextChange no matter the Openflow entries are installed on the OVS
+// bridge successfully or not.
+func (c *client) installConjunctiveMatchFlows(flowChanges []*conjMatchFlowContextChange) error {
+	// Send the Openflow entries to the OVS bridge.
+	if err := c.syncConjunctiveMatchFlows(flowChanges); err != nil {
+		return err
+	}
+	// Update conjunctiveMatchContext.
+	for _, ctxChange := range flowChanges {
+		ctxChange.updateContextStatus()
+	}
+	return nil
+}
+
+// syncConjunctiveMatchFlows sends all the changed Openflow entries to the OVS bridge in a single Bundle.
+func (c *client) syncConjunctiveMatchFlows(changes []*conjMatchFlowContextChange) error {
```

Comment:
Maybe `sendConjunctiveMatchFlows` is a better name, there isn't really a "sync" going on.

Reply:
@wenyingd I don't think you actually changed the name here?

Reply:
My fault, it seems I forgot to persist the in-memory changes. Updated.